### ISSUE TITLE
[BUG]S22i information not transmitted to QZ App

### ISSUE THREAD

**Issue by Maverick54400** [cagnulein/qdomyos-zwift#2878]
**Describe the bug**
I've installed the QZ companion app on my S22i bike, I've correctly entered the ip address under the proform tdf companion ip section but nothing happen when I launch a manual training on the bike. You can find the debug log attached

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11

**Smartphone (please complete the following information):**
 - Device: Samsung S24

**Append a debug log**
[debug-Thu_Dec_5_11_03_54_2024.log](https://github.com/user-attachments/files/18021441/debug-Thu_Dec_5_11_03_54_2024.log)

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
  
  
**Additional context**

**Comment by cagnulein** [comment]
hi @Maverick54400 the problem in the log is that i don't see a message at all from the companion. is it running on the bike? is the bike connected correctly over wifi? which device did you select on the companion?

**Comment by Maverick54400** [comment]
Hi,

thanks for your answer. Yes I can confirm that the companion is running :

![image](https://github.com/user-attachments/assets/7fa892d8-eed7-442b-8a92-174eeb9d21cf)

thanks for your support

**Comment by cagnulein** [comment]
What about my other questions?

**Comment by Maverick54400** [comment]
Oh sorry, yes my bike is correctly connected on my wifi and I ́ve selected s22i bike option on the companion app on the bike

**Comment by cagnulein** [comment]
does it have the new ifit beta 2.0 or the standard version?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 5 dic 2024 alle ore 13:29 Maverick54400 <
***@***.***> ha scritto:

**Comment by Maverick54400** [comment]
Here the details about my app 

![20241205_133759](https://github.com/user-attachments/assets/6f8343fc-aba5-453c-88fd-0e0df70387a0)

**Comment by cagnulein** [comment]
@victorypoint is this the new ifit version? in the meantime you can try to use this version that uses OCR https://github.com/cagnulein/QZCompanionNordictrackTreadmill/actions/runs/12175092604/artifacts/2277746109

download it and replace the apk into the install folder (you have to rename it)
at the start it will ask for recording your screen, confirm it there

let me know

**Comment by victorypoint** [comment]
@cagnulein I'm not sure.. need to see a pic of the main iFit screen or workout screen.

**Comment by Maverick54400** [comment]
Here a photo of the main ifit screen

![Uploading 20241205_141804.jpg…]()

**Comment by victorypoint** [comment]
@Maverick54400 looks like iFit v1. Can you upload a pic of the workout screen just to be sure?

**Comment by cagnulein** [comment]
@Maverick54400 did you install the companion using adb from a pc? if so we can try to select also the gran tour pro from the original companion and reboot the bike

**Comment by Maverick54400** [comment]
@victorypoint you're right I've checked on internet and it seems to be the V1 version.

@cagnulein yes I've installed the companion using adb from my pc

**Comment by cagnulein** [comment]
@Maverick54400 ok so try the gran tour device first and let's see if we see something on the qz debug log on the phone.
@victorypoint thanks

**Comment by Maverick54400** [comment]
here the logs. For information when I've choose the gran tour bike and launch the QZ app my ifit screen switch the screen every seconds (for example if I've launch a workout it switch from the workout screen to the pause screen every one second)
[debug-Thu_Dec_5_14_45_58_2024.log](https://github.com/user-attachments/files/18024202/debug-Thu_Dec_5_14_45_58_2024.log)

**Comment by cagnulein** [comment]
ok i can see now the messages from the companion, but they are only about gear and resistance.
so i guess the bike is not logging the metrics, ok so let's go to this plan https://github.com/cagnulein/qdomyos-zwift/issues/2878#issuecomment-2520219715

**Comment by Maverick54400** [comment]
Sorry but I don't understand where to put the apk file and how to do it. Can you explain me please?

**Comment by cagnulein** [comment]
pick that apk from the link, and put it in the insstallationfolder that you already have. renaming it to QZCompanionNordictrackTreadmill.apk

**Comment by Maverick54400** [comment]
Ok done but the companion app crash everytime I tried to launch it

**Comment by cagnulein** [comment]
@Maverick54400 can you try this instead https://github.com/cagnulein/QZCompanionNordictrackTreadmill/actions/runs/12097291287/artifacts/2256675112 ?

**Comment by Maverick54400** [comment]
same things with this one

**Comment by Maverick54400** [comment]
I think my bike has a problem, I prefer stop investigating on this issue and keep my bike as it is.

Thanks for your help anyway :)

**Comment by cagnulein** [comment]
i don't think it's a bike problem :) anyway if you will change your mind i'm here

**Comment by Maverick54400** [comment]
Hi,

Just to be sure, in the qz app I ve just to add the ip on the bike options as below ?

![Screenshot_20241206_100942](https://github.com/user-attachments/assets/66e820f7-c947-4e8b-9974-dc0b49f2fb10)

**Comment by cagnulein** [comment]
Yes @Maverick54400

**Comment by Maverick54400** [comment]
Thank you for your answer, I have reinstalled all on the bike and on my phone and I tried to launch the adb-debug :

![image](https://github.com/user-attachments/assets/9976e4e4-fa93-4e57-81c7-f789377ffc64)

Maybe that can help you understand, it seems to have a connection issue but the installation work great through adb connection

Thanks

**Comment by cagnulein** [comment]
Maybe the adb settings must be enabled again? It happens on some bikes

@victorypoint any hints?

**Comment by victorypoint** [comment]
@Maverick54400, yeah it sounds like ADB debugging maybe got turned off. Also make sure everything is on same wifi band and there is no VPN running on your PC.

**Comment by Maverick54400** [comment]
Hi,

I can confirm everything is on the same wifi without any vpn.

My qz app is in version 2.16 it’s the one directly accessible on the android play store. Maybe it’s not the good version of the app?

Thabks

**Comment by cagnulein** [comment]
so @Maverick54400 you still have that the companion is crashing on startup? did you try to roll back to the previous one and it's working again? (at least doesn't crash?)

**Comment by Maverick54400** [comment]
Yes I ve reinstalled the first one, the one available on the git hub website. 

This one seems working fine except that it doesn't send me any information in the qz app all the tiles remains at 0.

Thanks

**Comment by Maverick54400** [comment]
Yes adb is working to install the companion but it send me the error message below when I tried to use the adb-debug.bat :

![17338281025784320242937435228749](https://github.com/user-attachments/assets/072cd312-6be6-4bf8-b5a7-11bb07417200)

**Comment by cagnulein** [comment]
ok this is a good news it means there is a difference in the script.

i will try to compare them

@victorypoint do you have any idea?

**Comment by cagnulein** [comment]
@Maverick54400 download this file https://raw.githubusercontent.com/cagnulein/QZCompanionNordictrackTreadmill/refs/heads/master/InstallPackage/adb-debug.bat

and replace it into the install folder and try to run it again.

let's see if it fix the adb error

**Comment by Maverick54400** [comment]
ok with this new script the result is a little bit confusing. Failed to connect but it says adb connection successfull :D :

![image](https://github.com/user-attachments/assets/5359ed12-b145-4959-9dbe-7fb865e7d62d)

**Comment by victorypoint** [comment]
@cagnulein, okay that's strange. @Maverick54400, try the 'QZ-Companion-simple.bat' in the same folder and let us know the output.

**Comment by Maverick54400** [comment]
unfortunatly it's the same with QZ-companion-simple.bat :

![image](https://github.com/user-attachments/assets/64a43f7b-4cc3-46dd-9806-258e2e265e7d)

**Comment by victorypoint** [comment]
@Maverick54400, look for existing "ADB.exe" processes in task manager and kill them. Then try the companion scripts again.

**Comment by Maverick54400** [comment]
always the same, if I tried a telnet it's working fine, I don't know if it's usefull for you :

![image](https://github.com/user-attachments/assets/793a4cf5-56fe-4e4a-93b9-ef27b53496e8)

**Comment by victorypoint** [comment]
@Maverick54400, and ADB debugging is turned on?

**Comment by Maverick54400** [comment]
I don’t know, can you tell me hiw to verify or how to enabled it?

**Comment by cagnulein** [comment]
> @Maverick54400, and ADB debugging is turned on?

The full qz companion batch file, so I assume "yes"?

**Comment by victorypoint** [comment]
@Maverick54400, enable privileged mode on your bike tablet and check the ADB debugging setting under developer settings.

**Comment by Maverick54400** [comment]
I ve check and I only see usb debugging and it’s turning on. I didn’t see any adb debugging options on the developers option

**Comment by Maverick54400** [comment]
I don’t really remember when I bought it, I think more or less 2 or 3 years ago. I ve checked and the android version of the bike is android 7.1.2

**Comment by victorypoint** [comment]
Ok. Without usb debugging option we might be out of luck.

**Comment by Maverick54400** [comment]
Usb debugging is correctly enabled on the bike, I just didn’t find any adb debugging options

**Comment by victorypoint** [comment]
@Maverick54400, does the bike have a usb port? If so, have you tried a direct connection to your PC?

**Comment by Maverick54400** [comment]
No unfortunatly there is no usb port on the bike 😞

**Comment by cagnulein** [comment]
Tomorrow I will try to compare again the script. If the installation works it must work also the logcat

I will let you know

**Comment by Maverick54400** [comment]
Thanks for your time really appreciate 🙂

**Comment by cagnulein** [comment]
@Maverick54400 hello, i tried to compare the qzcompanion.bat and the adb debug.bat and they are identically from the adb point of view.

Just to be sure: without changing anything on the tablet side, if you start the qz-companion.bat are you able to connect and install the apk? If so can you share me the full log?

thanks

**Comment by Maverick54400** [comment]
Yes I'm able to install the apk and the companion app on the bike is correctly launched as background task. You can find the log below

[QZ-Companion-log.txt](https://github.com/user-attachments/files/18124869/QZ-Companion-log.txt)

**Comment by cagnulein** [comment]
@Maverick54400 i can't find any difference from the scripts. did you try to run the commands one by one without using the batchfile?

**Comment by cagnulein** [comment]
just to be sure i mean running this 

```
set /p TMIP="Enter treadmill IP address: "

ping %TMIP%
timeout 5

adb disconnect 
adb kill-server
adb connect %TMIP%

adb disconnect
adb kill-server
adb connect %TMIP%
adb connect %TMIP%
adb devices -l
```

you can copy one line by another and show me the full result

**Comment by Maverick54400** [comment]
ok here the full results of the commands : 

set /p TMIP="Enter treadmill IP address: "
Enter treadmill IP address: 192.168.10.75

ping %TMIP%

Envoi d’une requête 'Ping'  192.168.10.75 avec 32 octets de données :
Réponse de 192.168.10.75 : octets=32 temps=29 ms TTL=64
Réponse de 192.168.10.75 : octets=32 temps=36 ms TTL=64
Réponse de 192.168.10.75 : octets=32 temps=48 ms TTL=64
Réponse de 192.168.10.75 : octets=32 temps=56 ms TTL=64

Statistiques Ping pour 192.168.10.75:
    Paquets : envoyés = 4, reçus = 4, perdus = 0 (perte 0%),
Durée approximative des boucles en millisecondes :
    Minimum = 29ms, Maximum = 56ms, Moyenne = 42ms

adb disconnect
* daemon not running; starting now at tcp:5037
* daemon started successfully
disconnected everything

adb kill-server

adb connect %TMIP%
* daemon not running; starting now at tcp:5037
* daemon started successfully
failed to connect to 192.168.10.75:5555

adb disconnect
disconnected everything

adb kill-server

adb connect %TMIP%
* daemon not running; starting now at tcp:5037
* daemon started successfully
failed to connect to 192.168.10.75:5555

adb connect %TMIP%
already connected to 192.168.10.75:5555

adb devices -l
List of devices attached
192.168.10.75:5555     offline transport_id:1

**Comment by victorypoint** [comment]
@Maverick54400, "offline transport_id:1" indicates that your bike is connected to your PC, but ADB is unable to communicate with it. Here's a few things to try:
- Make absolute sure "USB Debugging" is enabled on the bike.
- Disable Windows firewall temporarily and try ADB connect again.
- Try a different Windows PC if you can.

**Comment by Maverick54400** [comment]
I can confirm that usb debugging is correctly enabled on the bike and Windows firewall is off on my computer.
I tried from a mac and the result is the same.
Thanks for your support

**Comment by cagnulein** [comment]
@victorypoint It must be something different because the full companion installation works

**Comment by victorypoint** [comment]
@cagnulein, interesting. So ADB must be working fine. Is it just ADB device control that is the issue here?

**Comment by Maverick54400** [comment]
My problem is that the tiles stay at 0 in the QZ app on my android phone.
I tried the app on my computer and it's the same the tiles doesn't move

**Comment by victorypoint** [comment]
@Maverick54400, @cagnulein, so if device is set to TDF Companion IP in QZ with correct IP, and device is set to S22i Bike in QZ Companion, then the issue is QZ isn't receiving UDP metrics from bike.

**Comment by cagnulein** [comment]
@Maverick54400  We are trying to investigate the new companion crash, that's why we need adb logs. @victorypoint

**Comment by Maverick54400** [comment]
Hi guys, good news. Today I've got the new ifit 2.0 update with the new interface and now tha app is working, the tiles are correctly updated.

I've made a little test and the last problem is that the inclinaison and the resistance are not changing from the app. Maybe a setting to change?

Thanks

**Comment by cagnulein** [comment]
That's great! @victorypoint do you have the same interface so now? In case I can use the same touch event.

@Maverick54400 can you share a screenshot of your manual window workout of ifit?

**Comment by Maverick54400** [comment]
Of course, here a screenshot of the new interface 

![20241220_232857](https://github.com/user-attachments/assets/3179a08f-75cf-4f59-b49d-3aeb33454749)

**Comment by cagnulein** [comment]
@victorypoint Is it the same?

**Comment by Maverick54400** [comment]
Unfortunatly, I ve made a reset of the bike to have a clean installation and I don’t have the new interface anymore.

I think I will need to wait until it propose me again the upgrade.

**Comment by cagnulein** [comment]
So the companion is crashing again?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 21 dic 2024 alle 10:16 Maverick54400 ***@***.***>
ha scritto:

**Comment by Maverick54400** [comment]
The companion is not crashing but the tiles are not updated in the qz app. It seems to have a problem with the old version of ifit

**Comment by cagnulein** [comment]
hi @Maverick54400 can you try to use this apk for the companion https://github.com/cagnulein/QZCompanionNordictrackTreadmill/actions/runs/12392144945/artifacts/2337179123
this will log directly on the qz app so i can see what's going on on the tablet
Thanks

**Comment by Maverick54400** [comment]
Hi, I've installed this version of the companion app but this one is crashing everytime I tried to launch it

![image](https://github.com/user-attachments/assets/f0637248-6b2d-4732-ad10-43475862a130)

**Comment by cagnulein** [comment]
But which companion did you install before that wasn't crashing?

**Comment by Maverick54400** [comment]
It was the companion app available on the github for the nordictrack configuration.

**Comment by cagnulein** [comment]
So not the one that I share with you 10 days ago here right?

If so we just go back to your original issue. Let's wait the ifit2 update

**Comment by Maverick54400** [comment]
No the one you give me is crashing everytime I try to launch it.

Yes I will waiting for the 2.0 update, I hope it will not be too long. It seems we can’t force it

**Comment by Maverick54400** [comment]
Hi, the 2.0 update is finally here this evening. So the tiles are correctly updated on the QZ App with the companion app (the one from the Nordicktrack github).

The last problem is the inclinaison and the resistance that are not updated on the bike from the QZ App. I think it's just a setting in the app, any ideas?

Thanks.

**Comment by cagnulein** [comment]
Do you mean you don't see the actual resistance and inclination value on Qz?
This is because I'm reading only the line with the metrics to speed up the
refresh rating


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 26 dic 2024 alle 20:56 Maverick54400 ***@***.***>
ha scritto:

**Comment by Maverick54400** [comment]
No if I set the inclinaison and the resistance on the bike I see the good value in the app.

But if I try to change the resistance and the inclinaison from the app nothing change on the bike.

**Comment by cagnulein** [comment]
Yes because I didn't add the engine to do so :)

If you press the inclination in qz don't you see any touch event on the
bike? I mean qz is sending a touch event to the bike probably to a wrong
place, don't you see it?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 26 dic 2024 alle 21:57 Maverick54400 ***@***.***>
ha scritto:

**Comment by Maverick54400** [comment]
No I didn't see anything move. I tried on a manual training on ifit.

I didn't try on Zwift yet. Is the QZ App should adapt automatically the inclinaison and the resistance on Zwift?

**Comment by cagnulein** [comment]
yes qz will adapt both from zwift but first we need to understand why you don't see a reaction. 
try to set the device x22i_noadb in the companion app and restart the bike.
then change the inclination tile without zwift, from qz. you should see a reaction. be sure to allow all the permission at the companion.
let me know

**Comment by Maverick54400** [comment]
I tried with x22 noadb settings and nothing is moving on the screen.

I tried on a manual training on ifit, do I need to try on zwift?

**Comment by cagnulein** [comment]
nope, are you sure you enabled all the permission for the companion in the android settings on the tablet?
if so are you able to access adb now on the new ifit version?

**Comment by Maverick54400** [comment]
I confirm that the companion app have all the permission. 

And I don’t have adb option on the android options on the bike

**Comment by cagnulein** [comment]
@victorypoint did you already put some notes about how to enable adb on the ifit 2? i guess it's your same bike so the steps should be the same, isn't it?

**Comment by cagnulein** [comment]
@Maverick54400 just to be sure, are you seeing this when you enable the noadb 
[Screen_recording_20241227_121056.webm](https://github.com/user-attachments/assets/e2677fb1-0bb2-4b8a-8663-ef0c7e5ac823)

**Comment by Maverick54400** [comment]
Yes that’s it

**Comment by cagnulein** [comment]
I mean do you see the accessibility window then? Did you allow it from there?

**Comment by Maverick54400** [comment]
Yes I see this screen pop out but it close immediatly I can’t do anything in it.

Since Ifit upgrade to 2.0 I can’t open the android settings anymore the screen reduce the window and I can’t open it.


https://github.com/user-attachments/assets/e4bc9c6c-535b-4af8-89b5-9b58eb947564


https://github.com/user-attachments/assets/29012720-215e-4b91-83fe-d272bb7a7892

**Comment by cagnulein** [comment]
@victorypoint did you see the same too? is there a workaround for this? this is the issue @Maverick54400

**Comment by Maverick54400** [comment]
Ok I find a workaround by desactivate the privilege mode and enabled it again. I have to do this each time because after the reboot the problem with the settings menu come back

I make the settings you told me in the accessibility menu but nothing change if I change inclinaison settings ine the qz app

**Comment by cagnulein** [comment]
@Maverick54400 could you please enable the setting in the developer setting in the bike's tablet where you can see the touch event with a mark? so we will be sure about the touch input. thanks

**Comment by Maverick54400** [comment]
Ok I found the culprit, it’s an app called glassOS which is installed with ifit 2.0 and prevent the access and change in the android settings 

I tried to uninstall it but it come back each time the bike is restarted. Moreover the qz companion accessibility setting you mak me enabled is come in back to unchecked at restart

![20241227_143200](https://github.com/user-attachments/assets/43d5df9d-b038-4444-ace6-fa1072fc9f1b)

**Comment by cagnulein** [comment]
i really hate nordictrack for this, they are really trying to force people to stay blocked to their subscriptions. so probably they are resetting the settings at each startup.
i guess we can't control the bike without adb and without accessibility. but @victorypoint did it, so there is for sure a way. let's wait for him

**Comment by victorypoint** [comment]
@cagnulein, @Maverick54400. Sorry for the late reply. We have a house full of family for Christmas and I don't get much time to get away. I've put together some notes and tools to help with install of QZ Companion. See attached. Let me know if you need further help. iFit 2.0 is a big pain in the a** but there are still workarounds to get metrics and auto-control working.

@cagnulein, these notes and tools might be good to include in the QZ Companion instructions. Look them over and make changes if required.

[ifit2-tools.zip](https://github.com/user-attachments/files/18261599/ifit2-tools.zip)

**Comment by Maverick54400** [comment]
Thanks for your update but unfortunatly nothing is working :

![image](https://github.com/user-attachments/assets/acc485ca-4a82-4fc4-96fb-9a5443a39fa5)

I tried to download a shell terminal on the bike tablet to execute the command locally but of course it's not possible to install an apk. I think Nordictrack is too powerfull.

Thanks for your help anyway :)

**Comment by cagnulein** [comment]
Ok so at least you have reading metrics, right @Maverick54400 ?

**Comment by Maverick54400** [comment]
yes I have reading metrics and I suppose I can't do more on my system.

I will see how it works with zwift.

**Comment by cagnulein** [comment]
Ok let me know 
I guess the @victorypoint solution it should work also for you since it's the same android build, maybe you should just understand which is the root issue

In case we can help you in the troubleshooting

**Comment by Maverick54400** [comment]
Hi, I tried Zwift and it's not working. I can select wahoo kickr 0003 as speed and cadence sensor but nothing happen when the training start

**Comment by cagnulein** [comment]
What do you mean? Are you starting the workout in the bike? Do you see the metrics in qz? On which device zwift is running?

**Comment by Maverick54400** [comment]
I start a manual workout on the bike, i see the metrics on the qz app but my avatar is not moving in zwift.

I use zwift on a Microsoft surface

**Comment by cagnulein** [comment]
I already heard about zwift issues on surface with arm. Is it the only device that you can try zwift? Did you press start on qz? I mean is not blinking red qz right?

**Comment by Maverick54400** [comment]
Yes I know Zwift is not working on Surface with ARM but mine is one with an Intel I5 processor. I don't have another device for test.

QZ app is correctly start and not on pause mode.

**Comment by victorypoint** [comment]
I also have a Surface Book 2 i5 and QZ and Zwift all work fine. So @Maverick54400 is ADB not working between your surface book and bike? Sorry if I asked this before.

**Comment by cagnulein** [comment]
@Maverick54400 don't you see the QZ device bluetooth as well in zwift along with the wahoo kickr? try to rename the qz phone to "bike" and reboot it if you don't see it

**Comment by Maverick54400** [comment]
I have the qz app on my samsung s24 phone, zwift on my surface computer both correctly connected.

Adb connection is strange between my bike and my surface because the command "adb install" works fine to install the companion app but all the commands "adb shell" don’t because it tell me adb device offline.
As you probably see, if I execute one of your script it tell me adb device offline, but if I launch the command adb connect it tells me that I am already connected the bike.

Maybe it can be the root cause of my issue.

**Comment by cagnulein** [comment]
> @Maverick54400 don't you see the QZ device bluetooth as well in zwift along with the wahoo kickr? try to rename the qz phone to "bike" and reboot it if you don't see it

@Maverick54400 did you do this?

**Comment by Maverick54400** [comment]
Yes I see my phone in the list. I tried to use the phone instead of the wahoo but the result is the same

**Comment by cagnulein** [comment]
@Maverick54400 try this:
- enable the android notification icon setting in the experimental settings in qz
- restart qz
- open zwift on the same android device
- are you able to see the metrcs on zwift then on the same device?

let me know, in this way must work so then in case it's a surface issue that we have to investigate

**Comment by Maverick54400** [comment]
Same thing with this configuration, I can choose the Wahoo kickr as cadence and power sensor but my avatar is not moving in the training

**Comment by cagnulein** [comment]
Can you share a fresh debug log from qz? I'm going to bed now, I will check it tomorrow 
Thanks, we are near

**Comment by Maverick54400** [comment]
Of course, here the logs 

[debug-Sat_Dec_28_22_46_39_2024.log](https://github.com/user-attachments/files/18267822/debug-Sat_Dec_28_22_46_39_2024.log)

Thanks for your help

**Comment by cagnulein** [comment]
In this log I can see only speed from the bike, that's the issue. Don't you
see the wattage too? When did this start?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 28 dic 2024 alle 22:58 Maverick54400 ***@***.***>
ha scritto:

**Comment by Maverick54400** [comment]
No the watts and the rpm didn’t move

**Comment by cagnulein** [comment]
but @Maverick54400 were they work previously? wattage is required for zwift, that's the reason. in case enable the debug log check on the companion so we can collect the reason on the debug log on the qz main app
let me know

**Comment by Maverick54400** [comment]
No the watts never move, only the speed and the inclinaison move.

I activate the dump log on the companion but I don’t know how to get it from the bike to send it to you

**Comment by cagnulein** [comment]
simply collect a new debug log from the phone, and they will be integrated on the log as well.
just to be sure: are you seeing wattage on the ifit manual view? this is mandatory of course

**Comment by Maverick54400** [comment]
Ok here the log and a photo with the watts on the bike screen

![20241229_110111](https://github.com/user-attachments/assets/7b469f52-cd99-4e89-b6e8-c49541ede7e1)

**Comment by cagnulein** [comment]
I don't see the log, thanks

**Comment by Maverick54400** [comment]
[debug-Sun_Dec_29_11_00_36_2024.log](https://github.com/user-attachments/files/18269141/debug-Sun_Dec_29_11_00_36_2024.log)

**Comment by cagnulein** [comment]
I don't see the log from the bike inside the log. On the bike, after enabling the debug log check, did you restart the bike? Can you show me a screenshot from the bike tablet of the companion app just to be sure?

**Comment by Maverick54400** [comment]
The dump log in the bike are desactivated each time I reboot the bike

![20241229_114053](https://github.com/user-attachments/assets/86f5b812-20e0-43f3-8481-ef16013373cb)

**Comment by cagnulein** [comment]
Hah sorry my fault, we were talking about 2 different version. I just create a new version of the companion with this log check.

You can download from here https://github.com/cagnulein/QZCompanionNordictrackTreadmill/releases/download/3.6.0/QZCompanionNordictrackTreadmill.apk

**Comment by Maverick54400** [comment]
here the logs with this new version :

[debug-Sun_Dec_29_11_57_07_2024.log](https://github.com/user-attachments/files/18269215/debug-Sun_Dec_29_11_57_07_2024.log)

**Comment by cagnulein** [comment]
I missed another thing, you need to enable the ocr and debug check on this companion app.

Sorry I'm following too many different tickets and I'm losing my mind

**Comment by Maverick54400** [comment]
With the OCR activated the tiles are not updated in the QZ App, I don't know if it's normal

[debug-Sun_Dec_29_12_07_10_2024.log](https://github.com/user-attachments/files/18269267/debug-Sun_Dec_29_12_07_10_2024.log)

**Comment by Maverick54400** [comment]
Moreover with the ocr activated the companion app is crashing everytime I launch it

**Comment by cagnulein** [comment]
Ok so went  back to the original issue where we have the crash with ocr. We
need to understand the root cause of the crash to fix this. When it crash
can't you see a way to see a log of the crash as android does?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 29 dic 2024 alle 12:44 Maverick54400 ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
Otherwise, if the speed on the bike increases if you increase the wattage proportionally we can create a algorithm to generate wattage from the speed

**Comment by Maverick54400** [comment]
No I can’t see any log on android as it’s an android with limited access

**Comment by cagnulein** [comment]
So check if there is a correlation between power and speed with inclination 0, somehow I can generate an algorithm to get power from speed if it's so

**Comment by bkeown64** [comment]
Looking at the manual workout photo that Maverick uploaded, he has iFit version 2.0, not v1.  The v1 version manual workout had a 2 dimensional racing track with a top view of the track. The track view shown in his photo is the angle view of the track which is on the latest iFit v2

**Comment by cagnulein** [comment]
yes he has both
